src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,170 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> next;
+        private Node<T> prev;
+
+        private Node(Node<T> prev, T value, Node<T> next) {
+            this.value = value;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> nodeToAdd = new Node<>(null, value, null);
+        if (size == 0) {
+            tail = nodeToAdd;
+            head = tail;
+            tail.prev = null;
+            tail.next = null;
+            size++;
+            return;
+        }
+        nodeToAdd.prev = tail;
+        tail.next = nodeToAdd;
+        tail = nodeToAdd;
+        size++;
+        return;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> nodeToAddByIndex = new Node<>(null, value, null);
+        if (index == size) {
+            add(value);
+            return;
+        } else if (index == 0) {
+            nodeToAddByIndex.next = head;
+            head = nodeToAddByIndex;
+            head.next.prev = head;
+            size++;
+            return;
+        } else {
+            Node<T> currentNode = getNodeByIndex(index);
+            nodeToAddByIndex.prev = currentNode.prev;
+            currentNode.prev.next = nodeToAddByIndex;
+            nodeToAddByIndex.next = currentNode;
+            currentNode.prev = nodeToAddByIndex;
+            size++;
+            return;
+        }
     }
```

Comment:
В этом методе мы можем либо просто убрать return statements либо упростить структуру блоков:
```suggestion
    public void add(T value, int index) {
        Node<T> nodeToAddByIndex = new Node<>(null, value, null);
        if (index == size) {
            add(value);
            return;
        } 
        if (index == 0) {
            nodeToAddByIndex.next = head;
            head = nodeToAddByIndex;
            head.next.prev = head;
            size++;
            return;
        } 
        Node<T> currentNode = getNodeByIndex(index);
        nodeToAddByIndex.prev = currentNode.prev;
        currentNode.prev.next = nodeToAddByIndex;
        nodeToAddByIndex.next = currentNode;
        currentNode.prev = nodeToAddByIndex;
        size++;
    }
```